File: Frontend/src/app/_interfaces/order-dto.ts
import { ProductOrderDto } from "./product-order-dto";

export interface OrderDto {
    id?: number;
    products: Array<ProductOrderDto>;
    address: string;
    comment?: string;
    price: number;
    delivererId?: string;
    customerId: string;
    orderStatus: string;
    deliveryTime?: Date;
}

File: Backend/Server/Server/DTOs/OrderDto.cs
using Server.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Server.DTOs
{
    public class OrderDto
    {
        public int? Id { get; set; }
        public IEnumerable<ProductOrderDto> Products { get; set; }
        public string Address { get; set; }
        public string? Comment { get; set; }
        public int Price { get; set; }
        public string? DelivererId { get; set; }
        public string CustomerId { get; set; }
        public OrderStatus OrderStatus { get; set; }
        public DateTime? DeliveryTime { get; set; }
    }
}
